File: py/283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

from typing import *

# @lc code=start


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[i], nums[left] = nums[left], nums[i]
                left+=1

# @lc code=end

File: 283.move-zeroes.cpp
/*
 * @lc app=leetcode id=283 lang=cpp
 *
 * [283] Move Zeroes
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        auto left = 0, right = 0;
        auto len = nums.size();
        while (right != len)
        {
            if (nums[right] != 0)
            {
                swap(nums[left++], nums[right++]);
            }
            else
            {
                ++right;
            }
        }
    }
};
// @lc code=end
